File: nat-kart-api/src/main/java/com/example/nat_kart_api/dto/HistoryDTO.java
package com.example.nat_kart_api.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class HistoryDTO {
    private int id;
    private RankingDTO player;
    private ConsoleDTO console;
    private CupsDTO cups;
    private int points;
    private boolean victory;
}

File: nat-kart/src/app/dto/historyDTO.ts
import { CupsDTO } from './cupsDTO';
import { RankingDTO } from './rankingDTO';
import { ConsoleDTO } from './consoleDTO';

export interface HistoryDTO {
  id?: number;
  player: RankingDTO;
  cups: CupsDTO;
  console: ConsoleDTO;
  points: number;
  victory: boolean;
}
